flexget/plugins/plugin_decompress.py

```diff
@@ -212,22 +214,39 @@ def handle_entry(self, entry, config):
                 os.makedirs(dest_dir)
 
             if not os.path.exists(destination):
+                success = False
+                error_message = ''
+                source = None
+
                 log.debug('Attempting to extract: %s to %s' % (path, destination))
                 try:
+                    # python 2.6 doesn't seem to like "with" in conjuntion with ZipFile.open
                     source = archive.open(path)
-                    target = file(destination, "wb")
-                    shutil.copyfileobj(source, target)
+                    with open(destination, 'wb') as target:
+                        shutil.copyfileobj(source, target)
+
                     log.verbose('Extracted: %s' % path )
-                except Exception as e:
+                    success = True
+                except (IOError, os.error) as e:
+                    error_message = 'OS error while creating file: %s (%s)' % \
+                                    (destination, e)
+                except (zipfile.BadZipfile, rarfile.Error) as e:
                     error_message = 'Failed to extract file: %s in %s (%s)' % \
                                     (path, archive_path, e)
+                except Exception as e:
+                    error_message = 'Unexpected error while extracting %s from %s (%s)' % \
+                                    (path, archive_path, e)
```

Comment:
We have a utility now to log a more detailed report for debugging if you want. Makes it easy for users to post debug logs and traceback should this happen. https://github.com/Flexget/Flexget/blob/develop/flexget/manager.py#L899